riomucho/__init__.py

```diff
@@ -61,77 +60,131 @@ def wrapper(*args, **kwds):
             return func(*args, **kwds)
         except Exception:
             raise MuchoChildError()
-
     return wrapper
 
 
-def main_worker(inpaths, g_work_func, g_args):
-    """TODO"""
-    global work_func
```

Comment:
This global isn't needed anymore, we now wrap the user function up in another class that gets passed to `imap_unordered`.